### ISSUE TITLE
CI failure after upgrade hexo-fs 0.2.3 to 1.0.0

### ISSUE THREAD

**Issue by YoshinoriN** [hexojs/hexo#3392]
## Detail

Some test failed caused by hexo-fs version up.
I tried to downgrade hexo-fs from 1.0.0 to 0.2.3. Test are pass after downgrade.

* [failed: master brunch](https://travis-ci.org/YoshinoriN/hexo/builds/468795796)
* [success: downgrade hexo-fs to 0.2.3](https://travis-ci.org/YoshinoriN/hexo/builds/469049346)

hexo-fs upgraded [2 month ago](https://github.com/hexojs/hexo/commit/6c9ea2b0cceaa4bc59724468cb7dddf8169e5a0e). But, test failed since 5 days ago. This time lag maybe caused by CI cache.

@segayuu, @tomap, @JLHwung , @tcrowe 
You committed or approval some changes between 0.2.3 and 1.0.0. 
Do you know anything about it? 

* [compare](https://github.com/hexojs/hexo-fs/compare/0.2.3...1.0.0)

**Comment by tcrowe** [comment]
I'm investigating these tests, @YoshinoriN, and I'll update you when I solve the problem.

**Comment by tcrowe** [comment]
Sometimes the package.json doesn't exist when it tries to update it:
https://github.com/hexojs/hexo/blob/master/lib/hexo/index.js#L168

Sometimes `hexo-fs#writeFile` does not write anything here:
https://github.com/hexojs/hexo/blob/master/lib/hexo/post.js#L77

I tried for a while to figure it out but I didn't come up with a solution yet. I'll try again soon.

**Comment by tcrowe** [comment]
Most tests remove the test file `after` it runs so it's difficult to see what went wrong:
https://github.com/hexojs/hexo/blob/master/test/scripts/box/box.js#L24
https://github.com/hexojs/hexo/blob/master/test/scripts/console/new.js#L34

**Comment by YoshinoriN** [comment]
Maybe this problem caused by [this commit](https://github.com/hexojs/hexo-fs/pull/23/commits/36c6aef19efbcae698fe8714f31984f5d477c948). And it change in `hexo-fs 1.0.1`.

@segayuu 
How do you think ? Do you have any ideas ? For example revert etc...

**Comment by NoahDragon** [comment]
@YoshinoriN you are right, when I revert that change, the test failures are gone. I will revert that change, and publish a new npm package.

**Comment by NoahDragon** [comment]
hexo-fs 1.0.2 has published. The change is only reverted the highlighted commit.
